### PR TITLE
Additional input type logic in elgg_view_input()

### PR DESCRIPTION
**Pull request by hypeJunction**
- Handle hidden input type
- Pass input type to field view

### CODE REVIEW

**Comment by hypeJunction** [comment]
Decided to not pass `input/hidden` through `elements/forms/input` view

File: engine/lib/views.php [hunk 1]
```diff
@@ -1320,6 +1324,7 @@ function elgg_view_input($input_type, array $vars = array()) {
 		'id' => $id,
 		'input' => $input,
 		'class' => $field_class,
+		'input_type' => $input_type,
```
Comment (juho-jaakkola) [inline engine/lib/views.php]: What is this used for?

Comment (hypeJunction) [inline engine/lib/views.php]: Nothing currently, but can be helpful if someone wants to do stuff in
hooks, e.g. add field class to view vars

**Comment by juho-jaakkola** [comment]
LGTM

**Pull request merged**